### PR TITLE
Mofnmodel

### PR DESCRIPTION
**Pull request by vasanthrajams**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json [hunk 1]
```diff
@@ -185,6 +185,23 @@
     }
   },
   "definitions": {
+    "MofNCertificateInfoObject": {
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json:188]: If we ever did generate a model, we don't use abbreviations in class names. You could either rename this to something more human readable, or leave the model name, but add an `x-ms-client-name`: https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-client-name

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json [hunk 2]
```diff
@@ -185,6 +185,23 @@
...
+      "properties": {
+        "Certificates": {
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json:190]: Property names are generally camelCase.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json [hunk 3]
```diff
@@ -226,7 +243,8 @@
     "SecurityDomainObject": {
       "properties": {
         "value": {
-          "type": "string"
+          "type": "string",
+          "description": "Security domain"
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json:247]: You should add `"format": "base64"` (or "base64url" for consistency with rest of KV, but either is fine) to clients know how to encode the data, if this is indeed the encoded data we discussed in our meeting.

**Review by heaths — APPROVED** [review]

**Review by heaths — APPROVED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json [hunk 4]
```diff
@@ -37,20 +37,20 @@
...
             "in": "body",
-            "name": "Certificates",
+            "name": "CertificateInfoObject",
             "required": true,
             "schema": {
-              "$ref": "#/definitions/CertificateSet"
+              "$ref": "#/definitions/CertificateInfoObject"
             },
-            "description": "Security domain download operation requires customer to provide three certificates containing public key in JWK format."
+            "description": "Security domain download operation requires customer to provide N certificates containing public key in JWK format."
```
Comment (amitbapat) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json:45]: Please specify minimum and maximum values of N - minimum 3, maximum 10.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json [hunk 5]
```diff
@@ -225,10 +232,113 @@
...
+                "key_algorithm": {
+                  "type": "string",
+                  "default": "shamir_share",
+                  "description": "The Algorithm used for shared keys"
+                },
+                "required": {
+                  "type": "integer",
+                  "minimum": 2,
+                  "maximum": 10,
+                  "description": "Number of keys required for security domain"
```
Comment (amitbapat) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json:253]: Please specify minimum and maximum values - minimum 2, maximum 10.

**Review by dcleblanc — COMMENTED** [review]
There would never be 2 pstore elements, there would be one pstore, and one for the partition backup.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json [hunk 6]
```diff
@@ -185,17 +185,24 @@
...
           },
           "minItems": 3,
-          "maxItems": 3,
+          "maxItems": 10,
           "uniqueItems": true,
-          "description": "Three certificates needed from customer"
+          "description": "Certificates needed from customer"
+        },
+        "quorum_threshold": {
+          "description": "Customer to specify the quorum number of certificates to restore security domain",
```
Comment (amitbapat) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json:201]: Add min and max values to description. Min=2, Max=10.

**Review by amitbapat — CHANGES_REQUESTED** [review]
Please include min and max limits in the description, so that auto-generated REST API doc looks good.

**Review by dcleblanc — COMMENTED** [review]
This sample is incorrect, not the same as the actual JSON

**Review by dcleblanc — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json [hunk 7]
```diff
@@ -222,8 +222,14 @@
     },
     "TransferKey": {
       "properties": {
+        "key_format": {
+          "type": "string",
+          "default": "jwk",
+          "description": "Specifies the format of the transfer key"
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json:228]: Should this be an enum, even if a value of only "jwk"? How do the customers know what to put here (especially as newer versions maybe support newer formats), or was this part of the opaque data that they shouldn't neeed to know about?

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json [hunk 8]
```diff
@@ -250,7 +256,7 @@
                   "type": "integer",
                   "minimum": 2,
                   "maximum": 10,
-                  "description": "Number of keys required for security domain"
+                  "description": "The number of keys (minimum 2 and maximum 10) required for security domain. "
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/securitydomain.json:259]: Wasn't the minimum 3 as mentioned above? Or is that different in this case (in which case, shouldn't the maximum be 9 then)?

**Review by heaths — APPROVED** [review]

**Pull request merged**